src/libmongoc/tests/test-mongoc-scram.c

```diff
@@ -130,6 +145,35 @@ test_mongoc_scram_sasl_prep (void)
    }
 #endif
 }
+
+static void
+test_mongoc_utf8_char_length (void)
+{
+   ASSERT_CMPINT (_mongoc_utf8_char_length (","), ==, 1u);
+   ASSERT_CMPINT (_mongoc_utf8_char_length ("ɶ"), ==, 2u);
+   ASSERT_CMPINT (_mongoc_utf8_char_length ("ྡྷ"), ==, 3u);
+   ASSERT_CMPINT (_mongoc_utf8_char_length ("🌂"), ==, 4u);
+}
+
+static void
+test_mongoc_utf8_string_length (void)
+{
+   ASSERT_CMPINT (_mongoc_utf8_string_length (",ase"), ==, 4u);
+   ASSERT_CMPINT (_mongoc_utf8_string_length ("ɸɴ"), ==, 2u);
+   ASSERT_CMPINT (_mongoc_utf8_string_length ("ྡྷ🌂e4🌕"), ==, 5u);
+   ASSERT_CMPINT (
+      _mongoc_utf8_string_length ("no special characters"), ==, 21u);
+}
+
+static void
+test_mongoc_utf8_to_unicode (void)
+{
+   ASSERT_CMPINT (_mongoc_utf8_get_first_code_point (",", 1), ==, 0x002C);
```

Comment:
```suggestion
   ASSERT_CMPUINT32 (_mongoc_utf8_get_first_code_point (",", 1), ==, 0x002C);
```
Use `ASSERT_CMPUINT32` to compare `uint32_t`.